pkg/chains/evm/proto_helpers.go

```diff
@@ -190,141 +240,241 @@ func ConvertCallMsgToProto(msg *evmtypes.CallMsg) (*CallMsg, error) {
-	for _, protoLog := range protoLogs {
-		logs = append(logs, convertLogFromProto(protoLog))
+	for i, protoLog := range protoLogs {
+		if protoLog == nil {
+			return nil, fmt.Errorf("log[%d] can't be nil", i)
+		}
+
+		l, err := convertLogFromProto(protoLog)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert log[%d] err: %w", i, err)
+		}
+		logs = append(logs, l)
 	}
-	return logs
+	return logs, nil
 }
 
-func convertLogFromProto(protoLog *Log) *evmtypes.Log {
+func convertLogFromProto(protoLog *Log) (*evmtypes.Log, error) {
+	if protoLog == nil {
+		return nil, fmt.Errorf("log can't be nil")
+	}
+
+	blockHash, err := ConvertHashFromProto(protoLog.GetBlockHash())
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert block hash err: %w", err)
+	}
+
+	topics, err := ConvertHashesFromProto(protoLog.GetTopics())
+	if err != nil {
+		return nil, errors.Join(ErrTopicsConversion, err)
+	}
+
+	eventSigs, err := ConvertHashFromProto(protoLog.GetEventSig())
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert event sigs err: %w", err)
+	}
+
+	address, err := ConvertAddressFromProto(protoLog.GetAddress())
+	if err != nil {
+		return nil, err
+	}
+
+	txHash, err := ConvertHashFromProto(protoLog.GetTxHash())
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert tx hash err: %w", err)
+	}
+
 	return &evmtypes.Log{
 		LogIndex:    protoLog.GetIndex(),
-		BlockHash:   evmtypes.Hash(protoLog.GetBlockHash()),
+		BlockHash:   blockHash,
 		BlockNumber: valuespb.NewIntFromBigInt(protoLog.GetBlockNumber()),
-		Topics:      ConvertHashesFromProto(protoLog.GetTopics()),
-		EventSig:    evmtypes.Hash(protoLog.GetEventSig()),
-		Address:     evmtypes.Address(protoLog.GetAddress()),
-		TxHash:      evmtypes.Hash(protoLog.GetTxHash()),
+		Topics:      topics,
+		EventSig:    eventSigs,
+		Address:     address,
+		TxHash:      txHash,
 		Data:        protoLog.GetData(),
 		Removed:     protoLog.GetRemoved(),
-	}
+		// TODO TxIndex
+	}, nil
 }
 
-func ConvertTopicsFromProto(protoTopics []*Topics) [][]evmtypes.Hash {
+func ConvertTopicsFromProto(protoTopics []*Topics) ([][]evmtypes.Hash, error) {
+	if protoTopics == nil {
+		return nil, fmt.Errorf("topics can't be nil")
+	}
+
 	topics := make([][]evmtypes.Hash, 0, len(protoTopics))
-	for _, topic := range protoTopics {
-		hash := make([]evmtypes.Hash, 0, len(topic.GetTopic()))
-		for _, t := range topic.GetTopic() {
-			hash = append(hash, evmtypes.Hash(t))
+	for i, protoTopic := range protoTopics {
+		if protoTopic == nil {
+			return nil, fmt.Errorf("topic[%d] can't be nil", i)
 		}
-		topics = append(topics, hash)
-	}
-	return topics
-}
 
-func ConvertLogToProto(log *evmtypes.Log) *Log {
-	var topics [][]byte
-	for _, topic := range log.Topics {
-		topics = append(topics, topic[:])
+		hashes, err := ConvertHashesFromProto(protoTopic.GetTopic())
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert topics[%d]: %w", i, err)
```

Comment:
same as the others, why printing the index instead of the offending value? (actually, in this case, if you change the function `ConvertHashesFromProto()` to print the value, then maybe the offset is ok here (but again, I'd just print the wrong value).
Also, can you group all the ones failing to report all at once?